components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -1817,7 +1817,13 @@ void AdsServiceImpl::OnPrefsChanged(
   if (pref == prefs::kEnabled ||
       pref == brave_rewards::prefs::kBraveRewardsEnabled) {
     if (IsEnabled()) {
-      RemoveOnboarding();
+#if !defined(OS_ANDROID)
+      if (first_run::IsChromeFirstRun()) {
+        SetBooleanPref(prefs::kShouldShowOnboarding, false);
+      } else {
+        RemoveOnboarding();
+      }
+#endif
```

Comment:
As the original code (presumably) affected Android as well, and now this is excluded - what _should_ / does happen on Android now?

Reply:
For Android the original code just exited inside `RemoveOnboarding` as onboarding is handled differently for Android. There is a ticket to refactor Android to use the same business logic flags for onboarding

Reply:
but now `RemoveOnboarding` is not triggered as the whole thing is wrapped for not android

Reply:
RemoveOnboarding does not need to be triggered for Android as they handle their own onboarding at this time

Reply:
@tmancey Provided Android will never rely on `RemoveOnboarding` being called in the future (as now it won't), I'm happy.